patroni/postgresql.py

```diff
@@ -813,6 +813,8 @@ def check_startup_state_changed(self):
                 # Let the main loop of run cycle clean up the mess.
                 logger.warning("%s status returned from pg_isready",
                                "Unknown" if ready == STATE_UNKNOWN else "Invalid")
+                return False
```

Comment:
Docstring tells following:
```python
         :returns: True iff state was changed from 'starting'
```

In this case state obviously changed, because pg_isready doesn't return STATE_REJECT anymore.
That means we can't return `False` here. 

I believe @ants had very good reasons for such implementation.